include/aspect/simulator/assemblers/advection.h

```diff
@@ -0,0 +1,102 @@
+/*
+  Copyright (C) 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_simulator_assemblers_advection_h
+#define _aspect_simulator_assemblers_advection_h
+
+
+#include <aspect/simulator/assemblers/interface.h>
+#include <aspect/simulator_access.h>
+
+namespace aspect
+{
+  namespace Assemblers
+  {
+    /**
+     * This class assembles the terms for the matrix and right-hand-side of the advection
+     * equation for the current cell.
+     */
+    template <int dim>
+    class AdvectionSystem : public Assemblers::Interface<dim>,
```

Comment:
One point that confused me about this design, and I guess we can talk about that, is that now, if you call an assembler, you don't know any more if that assembler changes the Stokes or the Advection system. Before, with the signals, they were all in one class that either said advection or Stokes or other or whatever, but now they are just in different files. So if the assembler is just called `AdvectionSystem`, that's totally fine, but if it's called something weird like `PhaseBoundaryAssembler`, it's suddenly not clear anymore what it does. 
So I guess we can either say we don't care, or we may want to think about some naming convention, or maybe you can think of another way for how to make this clearer (I think, in particular as the set_assemblers function is quite long, it could make sense to have names that are easy to understand).

Reply:
As discussed, one could also use different namespaces for the Stokes/Advection/Other assemblers. Then, the assembler would only have to be called `IncompressibleTerms` instead of `StokesIncompressibleTerms`, and would be in the Stokes namespace.

Reply:
I like the namespace idea (disclaimer: it was mine) :smile:. If nobody objects, I will regroup the assemblers accordingly.